docs/key-registry.md

```diff
@@ -0,0 +1,93 @@
+# Key Registry
+
+## Sequence Diagram
+
+TODO: add sequence diagram
```

Comment:
Are you planning to do this in this PR?

Reply:
I actually have one ready to go, just forgot to add it.